src/hooks/use-chain-id.hook.ts

```diff
@@ -0,0 +1,22 @@
+import { ChainIds } from '@taquito/taquito';
+import { noop } from 'lodash-es';
+import { useEffect, useState } from 'react';
+
+import { useSelectedRpcUrlSelector } from 'src/store/settings/settings-selectors';
+import { READ_ONLY_SIGNER_PUBLIC_KEY, READ_ONLY_SIGNER_PUBLIC_KEY_HASH } from 'src/utils/env.utils';
+import { createReadOnlyTezosToolkit } from 'src/utils/rpc/tezos-toolkit.utils';
+
+export const useChainId = () => {
```

Comment:
Actually, `chainId` is already cached in https://github.com/madfish-solutions/templewallet-mobile/blob/TW-666-deposit-part/src/utils/rpc/fast-rpc.ts#L261 . Due to memoization of `createReadOnlyTezosToolkit` function, chain ID may be re-requested only in case of changing RPC URL or switching an account. The latter happens pretty rarely, so let's neglect it